configure.ac

```diff
@@ -232,6 +235,47 @@ if test x$ac_cv_func_mkstemp = xyes ; then
 	AC_DEFINE(HAVE_MKSTEMP, 1, [Have mkstemp])
 fi
 
+# std::filesystem::path breaks compilation on OSX Mojave
+# or earlier versions of Xcode, despite the fact it's there
+AC_MSG_CHECKING([if std::filesystem::path compiles fine])
+AC_TRY_COMPILE([
+#if HAVE_FILESYSTEM
+#include <filesystem>
+#endif
+],
+[
+	std::filesystem::path a{};
+],
+	ac_cv_class_fs_path=yes,
+	ac_cv_class_fs_path=no)
+AC_MSG_RESULT($ac_cv_class_fs_path)
+AM_CONDITIONAL([HAVE_FILESYSTEM], [ test "x$ac_cv_class_fs_path" = xyes ])
+if test "x$ac_cv_class_fs_path" = xyes; then
+	AC_DEFINE(USE_STDFILESYSTEM, 1, [Use std::filesystem])
+fi
+
+AC_MSG_CHECKING([if std::filesystem requires linking stdc++fs])
+AC_TRY_LINK([
+#if HAVE_FILESYSTEM
+#include <filesystem>
+#endif
+#include <system_error>
+],
+[
+    std::error_code err;
+    auto a = std::filesystem::status("a", err);
```

Comment:
Probably need to use a symbol that won't come inlined from the headers, but from libstdc++-fs.

Reply:
@marzojr sure, but which one? Theoretically this should be `filesystem::path`... but well apparently it's not. Seems like it's going to be try-and-error until it works. If it works.

Reply:
Turns out that I made a mistake in variable naming, shame on me. -_-
@marzojr I made it fail to compile (as it should), now all the platforms on Travis build fine. Please let me know if the way it is fine.